src/collection-view.js

```diff
@@ -125,11 +125,86 @@ const CollectionView = Backbone.View.extend({
     }
   },
 
-  // get the child view by model it holds, and remove it
-  _onCollectionRemove(model) {
-    const view = this.children.findByModel(model);
-    this.removeChildView(view);
-    this._checkEmpty();
+  // Handle collection update model removals
+  _onCollectionUpdate(collection, options) {
+    const changes = options.changes;
+    if (changes) {
+      const removed = changes.removed;
+      if (_.isArray(removed)) {
+        this._removeChildViews(removed);
+      }
+    }
+  },
+
+  // Remove the child views and destroy them.
+  // This function also updates the indices of later views
+  // in the collection in order to keep the children in sync with the collection.
+  // Return array of views where the first view has the index that
+  // tells CollectionView#_updateIndices where to start syncing remaining
+  // view indices.
+  _removeChildViews(modelsOrViews, {checkEmpty} = {}) {
```

Comment:
maybe add a comment to explain the naming of the var `modelsOrViews` might be confusing